flang/lib/Semantics/check-cuda.cpp

```diff
@@ -413,4 +415,18 @@ void CUDAChecker::Enter(const parser::CUFKernelDoConstruct &x) {
   }
 }
 
+void CUDAChecker::Enter(const parser::AssignmentStmt &x) {
+  const evaluate::Assignment *assign = semantics::GetAssignment(x);
```

Comment:
Please always use the safer form of initialization with braces in semantics.

Reply:
Done